test/Microsoft.IdentityModel.Tokens.Tests/SupportedAlgorithmTests.cs

```diff
@@ -170,6 +171,24 @@ public static TheoryData<SupportedAlgorithmTheoryData> IsSupportedAlgorithmAndKe
                     "X509_CustomCryptoProviderFactory",
                     theoryData);
 
+                // Add "none" and "noNe" test cases for all key types
+                var keyTypes = new Dictionary<string, SecurityKey>
+                {
+                    { "JsonWebKey_Ecdsa", KeyingMaterial.JsonWebKeyP256 },
+                    { "JsonWebKey_Rsa", KeyingMaterial.JsonWebKeyRsa_2048 },
+                    { "JsonWebKey_Symmetric", KeyingMaterial.JsonWebKeySymmetric256 },
+                    { "Rsa", KeyingMaterial.RsaSecurityKey_2048 },
+                    { "Symmetric", KeyingMaterial.DefaultSymmetricSecurityKey_256 },
+                    { "X509", KeyingMaterial.X509SecurityKeySelfSigned2048_SHA256 },
+                    { "Ecdsa", KeyingMaterial.Ecdsa384Key }
+                };
+
+                foreach (var keyType in keyTypes)
+                {
+                    SupportedAlgorithmTheoryData.AddTestCase("none", keyType.Value, false, $"{keyType.Key}_none", theoryData);
```

Comment:
If we are trying to test casing, we can test a few more:

foreach (var variation in ["none", "None", "nOne", "noNe", "nonE",])

Reply:
+`"NONE"`? :)